src/main/java/org/apache/sling/scripting/core/impl/DefaultSlingScript.java

```diff
@@ -732,7 +735,16 @@ Bindings verifySlingBindings(final SlingBindings slingBindings) throws IOExcepti
 
             ProtectedBindings protectedBindings = new ProtectedBindings(bindings, protectedKeys);
             for (BindingsValuesProvider provider : bindingsValuesProviders) {
+                long start = System.nanoTime();
                 provider.addBindings(protectedBindings);
+                long stop = System.nanoTime();
+                LOGGER.trace("Invoking addBindings() of {} took {} nanoseconds",
+                        provider.getClass().getName(), stop-start);
+                if (stop-start > WARN_LIMIT_FOR_BVP_NANOS) {
+                    LOGGER.info("Adding the bindings of {} took {} microseconds; "
```

Comment:
nitpick, this message could say "...which is above the built-in limit of $WARN_LIMIT_FOR_BVP_NANOS"

Reply:
implemented